books_project/books_app/components/book_component.py

```diff
@@ -0,0 +1,63 @@
+from books_app.repositories.book_repository import BookRepository
+from rest_framework.exceptions import NotFound, ValidationError
+
+from books_app.repositories.author_repository import AuthorRepository
+
+class BookComponent:
+
+    def create_book(self, title, published_year, genr, isbn, author_id):
+        exists = BookRepository.check_isbn_exist(isbn)
+        author = AuthorRepository.get_author(author_id=author_id)
+        if not author:
+            raise NotFound("author not found", code=404)
+        elif title != "":
```

Comment:
you meant == here? either way we prefer to use "elif not title:". 
"not title" here could mean title is empty string, or title is null.